test/test_api_integration.py

```diff
@@ -0,0 +1,138 @@
+import http.client
+import os
+import unittest
+from unittest import mock
+
+import requests
+
+from xai_grok.grok import Grok
+from xai_grok import errors
+from xai_grok import providers
+from xai_grok import schemas
+
+
+class TestAPIIntegration(unittest.TestCase):
+
+    def setUp(self):
+        if xai_api_key := os.environ["XAI_API_KEY"]:
+            self.grok = Grok(
+                xai_api_key, requests, providers.UnauthorizedUserErrorProvider()
+            )
+        else:
+            raise Error(
+                "Environment variable XAI_API_KEY not set. Failed to instantiate xAI Grok client. Please get an xAI API key from https://console.x.ai"
+            )
```

Comment:
The `Error` exception is used but not imported. This will cause a `NameError` when the environment variable is not set. It should be replaced with a proper exception like `ValueError` or imported.